ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/BufferedWriteChannel.java

```diff
@@ -91,23 +98,26 @@ void flush() throws IOException {
     }
   }
 
-  CompletableFuture<Void> asyncFlush(ExecutorService executor) throws IOException {
+  CompletableFuture<Long> asyncFlush(ExecutorService executor, Long writtenIndex) throws IOException {
```

Comment:
Let's don't pass `writtenIndex`, which is not used anywhere except for returning back in the future.   It makes the method complicated.